homeassistant/components/reolink/select.py

```diff
@@ -0,0 +1,120 @@
+"""This component provides support for Reolink select entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import DayNightEnum, Host, SpotlightModeEnum
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkSelectEntityDescriptionMixin:
+    """Mixin values for Reolink select entities."""
+
+    method: Callable[[Host, int, str], Any]
+
+
+@dataclass
+class ReolinkSelectEntityDescription(
+    SelectEntityDescription, ReolinkSelectEntityDescriptionMixin
+):
+    """A class that describes select entities."""
+
+    supported: Callable[[Host, int], bool] = lambda api, ch: True
+    value: Callable[[Host, int], str] | None = None
+    get_options: Callable[[Host, int], Any] | None = None
+
+
+SELECT_ENTITIES = (
+    ReolinkSelectEntityDescription(
+        key="floodlight_mode",
+        name="Floodlight mode",
+        icon="mdi:spotlight-beam",
+        translation_key="floodlight_mode",
+        options=[mode.name for mode in SpotlightModeEnum],
+        supported=lambda api, ch: api.supported(ch, "floodLight"),
+        value=lambda api, ch: SpotlightModeEnum(api.whiteled_mode(ch)).name,
+        method=lambda api, ch, name: api.set_whiteled(ch, mode=name),
+    ),
+    ReolinkSelectEntityDescription(
+        key="day_night_mode",
+        name="Day night mode",
+        icon="mdi:theme-light-dark",
+        translation_key="day_night_mode",
+        options=[mode.name for mode in DayNightEnum],
```

Comment:
Since you're providing translations for this I think it would be a good idea that you have a translation dictionary so the key values doesn't change if the reolink API changes down the line

Reply:
Same as above, see https://github.com/starkillerOG/reolink_aio/blob/10c4bbd868d7e1f8cce4167d5098214100e01a26/reolink_aio/enums.py#L14-L19